f2/src/utils/scanFiles.js

```diff
@@ -76,34 +78,59 @@ const contentModerateFile = (file, callback) => {
         message: 'Error in Content Moderator',
         error: err,
       })
-      file[1].adultClassificationScore = 'Could not scan'
+      file.adultClassificationScore = 'Could not scan'
     } else {
       try {
         const contMod = JSON.parse(response.body)
-        file[1].isImageRacyClassified = contMod.IsImageRacyClassified
-        file[1].isImageAdultClassified = contMod.IsImageAdultClassified
-        file[1].adultClassificationScore = contMod.AdultClassificationScore
-        file[1].racyClassificationScore = contMod.RacyClassificationScore
+        file.isImageRacyClassified = contMod.IsImageRacyClassified
+        file.isImageAdultClassified = contMod.IsImageAdultClassified
+        file.adultClassificationScore = contMod.AdultClassificationScore
+        file.racyClassificationScore = contMod.RacyClassificationScore
       } catch (error) {
         console.warn(`Error in Content Moderator: ${error.stack} `)
       }
     }
-    callback(null, file[1])
+    callback(null, file)
   })
 }
 
+//Queue to call content moderator once per second.
+const queue = async.queue((task, callback) => {
```

Comment:
This queue object would a global object. Accessible by all requests

Reply:
Yes, if we want the app to only call the server once per second we will need to use the same queue. Otherwise overlapping requests could cause requests less than one second apart.